### DISCUSSION TITLE
ArgoCD and cluster upgrades with blue/green switch

### DISCUSSION THREAD

**Discussion by KevinDW-Fluxys** [argoproj/argo-cd#16088] [Q&A] [answered]
Hi All,

We recently started using ArgoCD and are wondering about managing it in our environment.
As part of our cluster upgrade strategy, we spin up a second cluster, install all applications and once it is ready, redirect all traffic to that cluster.

Now we have added ArgoCD as an application on one of our clusters. Our ArgoCD instance is managing multiple clusters. 
For our cluster upgrade this would mean that during the time of a cluster upgrade, there would be 2 instances of argocd running, looking at the same repositories and syncing them with the same clusters.

Would this cause conflicts during the time both instances are running? Is there a way to deal with this situation or should we take another approach?

Any other ideas or suggestions are welcome!

**Comment by FernandoMiguel** [comment, marked as answer]
based on our testing/experience, if both argos are the same version and config, there should be no problem for both argos to handles both set of clusters.

but if you want, you can have two application sets, and each cluster only loads one

**Reply by KevinDW-Fluxys** [reply]
We've done some switches now with both ArgoCDs running, and we haven't encountered any issues, so having multiple argoCDs running on the same config is a working solution for us
